python/mujoco/usd/shapes.py

```diff
@@ -14,34 +14,244 @@
 # ==============================================================================
 """Built-in shapes for USD exporter."""
 
-from typing import Dict, Any
+from typing import Dict, Any, Tuple, Optional, Union
 
 import mujoco
 import numpy as np
-from open3d import open3d as o3d
 
-
-def create_hemisphere(
-    radius: float, theta_steps: int = 50, phi_steps: int = 50
+def get_triangle_uvs(
+    vertices: np.array,
+    triangles: np.array,
+    texture_type: Optional[mujoco.mjtTexture]
 ):
-  """Creates a hemisphere mesh from a point cloud."""
-  points = []
-  for i in range(phi_steps + 1):
-    phi = np.pi / 2 * i / phi_steps
-    for j in range(theta_steps + 1):
-      theta = 2 * np.pi * j / theta_steps
-      x = radius * np.sin(phi) * np.cos(theta)
-      y = radius * np.sin(phi) * np.sin(theta)
-      z = radius * np.cos(phi)
-      points.append([x, y, z])
+  if texture_type == None:
+    return None
+
+  triangle_uvs = []
+  if texture_type == mujoco.mjtTexture.mjTEXTURE_2D:
+    triangle_uvs = [[vertices[i][0], vertices[i][1]] for i in np.nditer(triangles)]
+    
+  elif texture_type == mujoco.mjtTexture.mjTEXTURE_CUBE:
+    center = np.mean(vertices, axis=0)
+    for vertex_id in np.nditer(triangles):
+      x, y, z = vertices[vertex_id] - center
+
+      abs_x, abs_y, abs_z = abs(x), abs(y), abs(z)
+      
+      if x > 0 and abs_x >= abs_y and abs_x >= abs_z:
+        u = -z / abs_x
+        v = y / abs_x
+      elif x <= 0 and abs_x >= abs_y and abs_x >= abs_z:
+        u = z / abs_x
+        v = y / abs_x
+      elif y > 0 and abs_y >= abs_x and abs_y >= abs_z:
+        u = x / abs_y
+        v = -z / abs_y
+      elif y <= 0 and abs_y >= abs_x and abs_y >= abs_z:
+        u = x / abs_y
+        v = z / abs_y
+      elif z > 0 and abs_z >= abs_x and abs_z >= abs_y:
+        u = x / abs_z
+        v = y / abs_z
+      elif z <= 0 and abs_z >= abs_x and abs_z >= abs_y:
+        u = -x / abs_z
+        v = y / abs_z
+
+      u = (u + 1.0) / 2.0
+      v = (v + 1.0) / 2.0
+      v /= 6
+
+      assert 0 <= u and u <= 1 and 0 <= v and v <= 1
+
+      triangle_uvs.append([u, v])
+  elif texture_type == mujoco.mjtTexture.mjTEXTURE_SKYBOX:
+    # defaults to 2D mapping temporarily
+    triangle_uvs = [[vertices[i][0], vertices[i][1]] for i in np.nditer(triangles)]
+
+  return np.array(triangle_uvs)
+
+class TriangleMesh():
+  """ Store UV and geometry information for a primitve mesh
```

Comment:
Remove space after `"""`, fix "primitve" typo, add a period at the end of the sentence and move the closing `"""` up to this line.